From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nobuyoshi Nakada <nobu@ruby-lang.org>
Date: Sat, 23 Jan 2021 09:24:52 +0900
Subject: [PATCH] vcs.rb: continue just with warning if failed to fetch notes

---
 tool/lib/vcs.rb | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/tool/lib/vcs.rb b/tool/lib/vcs.rb
index fcee91a7a4cde7b1d08cddb41a584d49fd2a3f8d..6dafc6c426a36697598763e54ed79eddae9a7eda 100644
--- a/tool/lib/vcs.rb
+++ b/tool/lib/vcs.rb
@@ -606,12 +606,13 @@ def export_changelog(url, from, to, path)
         warn "no starting commit found", uplevel: 1
         from = nil
       end
-      unless svn or system(*%W"#{COMMAND} fetch origin refs/notes/commits:refs/notes/commits",
+      if svn or system(*%W"#{COMMAND} fetch origin refs/notes/commits:refs/notes/commits",
                            chdir: @srcdir, exception: false)
-        abort "Could not fetch notes/commits tree"
-      end
-      system(*%W"#{COMMAND} fetch origin refs/notes/log-fix:refs/notes/log-fix",
+        system(*%W"#{COMMAND} fetch origin refs/notes/log-fix:refs/notes/log-fix",
                chdir: @srcdir, exception: false)
+      else
+        warn "Could not fetch notes/commits tree", uplevel: 1
+      end
       to ||= url.to_str
       if from
         arg = ["#{from}^..#{to}"]
